server/datastore/mysql/hosts.go

```diff
@@ -1909,3 +1910,34 @@ func (ds *Datastore) ListHostBatteries(ctx context.Context, hid uint) ([]*fleet.
 	}
 	return batteries, nil
 }
+
+// countHostNotResponding counts the hosts that haven't been submitting results for sent queries.
+//
+// Notes:
+//   - We use `2 * interval`, because of the artificial jitter added to the intervals in Fleet.
+//   - Default values for:
+//     - host.DistributedInterval is usually 10s.
+//     - svc.config.Osquery.DetailUpdateInterval is usually 1h.
+//   - Count only includes hosts seen during the last 7 days.
+func countHostsNotRespondingDB(ctx context.Context, db sqlx.QueryerContext, config config.FleetConfig) (int, error,
+) {
+	interval := config.Osquery.DetailUpdateInterval.Seconds() * 2
+
+	// The `WHERE` clause is intended to capture where Fleet hasn't received a distributed write
+	// from the host during the interval since the host was last seen. Thus we assume the host
+	// is having some issue in executing distributed queries or sending the results.
+	sql := `
+	SELECT
+	  COUNT(*)
+	FROM (SELECT * FROM hosts JOIN host_seen_times ON hosts.id = host_seen_times.host_id) h
+	WHERE 
+	  TIME_TO_SEC(TIMEDIFF(h.seen_time, h.detail_updated_at)) >= GREATEST(h.distributed_interval, ?)
+	  AND TIMEDIFF(DATE_ADD(h.seen_time, INTERVAL 7 DAY), NOW()) > 0
+`
+
+	var count int
+	if err := sqlx.GetContext(ctx, db, &count, sql, interval); err != nil {
```

Comment:
Is it possible to instead get the `hostIDs`, log them (if len > 0) and return `len(hostIDs)`?
(See `Additional logging for troubleshooting` section of #6448.)